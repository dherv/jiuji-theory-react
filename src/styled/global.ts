import { createGlobalStyle } from "styled-components";

const GlobalStyle = createGlobalStyle`
:root {
  color: #484848
}
  body {
    font-family: Open-Sans, Helvetica, Sans-Serif;
    color: var(--color)
  }
`;

export default GlobalStyle;
