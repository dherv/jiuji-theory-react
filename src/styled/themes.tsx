export const themeBlue = {
  backgroundColor: 'rgba(18, 70, 246, 0.10)',
  color: '#1246F6',
  border: '1px solid transparent',
};

export const themeBlueLight = {
  backgroundColor: '#fff',
  color: 'rgba(18, 70, 246, 0.5)',
  border: '1px solid rgba(18, 70, 246, 0.5)',
  hoverBackgroundColor: 'rgba(18, 70, 246, 0.10)',
  hoverColor: '#1246F6',
  hoverBorder: '1px solid transparent',
};

export const themeBlueField = {
  border: '1px solid rgba(18, 70, 246, 0.1)',
};

export const themeBlueHeader = {
  color: '#fff',
  backgroundColor: 'rgba(18, 70, 246, 0.65)',
};

export const themeWhite = {
  color: '#fff',
  border: '1px solid #fff',
  fontWeight: 300,
};
