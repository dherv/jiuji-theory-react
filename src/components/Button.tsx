import styled from 'styled-components';
import { themeBlue } from '../styled/themes';

const Button = styled.button`
  padding: 0.5rem 1rem;
  font-size: 0.75rem;
  font-weight: 600;
  color: ${(props) => props.theme.color};
  background-color: ${(props) => props.theme.backgroundColor};
  border: ${(props) => props.theme.border};
  transition: 0.5s;
  &:hover {
    background-color: ${(props) => props.theme.hoverBackgroundColor};
    border: ${(props) => props.theme.hoverBorder};
  }
`;

Button.defaultProps = {
  theme: themeBlue,
};

export default Button;
