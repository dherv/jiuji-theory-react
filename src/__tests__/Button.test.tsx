import React from 'react';
import { fireEvent, render, screen } from '@testing-library/react';
import Button from '../components/Button';
import { themeBlueLight } from '../styled/themes';

const props = {
  onClick: jest.fn(),
};

test("should display a button with the text", () => {
  render(<Button>test</Button>);
  expect(screen.getByText("test")).toBeInTheDocument();
});

test("should render with the default them", () => {
  const { asFragment } = render(<Button>test</Button>);
  expect(asFragment()).toMatchSnapshot();
});

test("should render the theme light", () => {
  const { asFragment } = render(<Button theme={themeBlueLight}>test</Button>);
  expect(asFragment()).toMatchSnapshot();
});

test("should render the theme light hover on mouseOver", () => {
  const { asFragment } = render(<Button theme={themeBlueLight}>test</Button>);
  fireEvent.mouseOver(screen.getByText("test"));
  expect(asFragment()).toMatchSnapshot();
});

test("should call onClick", () => {
  render(<Button {...props}>test</Button>);
  fireEvent.click(screen.getByText("test"));
  expect(props.onClick).toHaveBeenCalled();
});
