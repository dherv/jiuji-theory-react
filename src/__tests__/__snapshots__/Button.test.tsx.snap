// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render the theme light 1`] = `
<DocumentFragment>
  <button
    class="sc-AxjAm jcymSN"
  >
    test
  </button>
</DocumentFragment>
`;

exports[`should render the theme light hover on mouseOver 1`] = `
<DocumentFragment>
  <button
    class="sc-AxjAm jcymSN"
  >
    test
  </button>
</DocumentFragment>
`;

exports[`should render with the default them 1`] = `
<DocumentFragment>
  <button
    class="sc-AxjAm dmAhUC"
  >
    test
  </button>
</DocumentFragment>
`;
